client/components/StartGameDialog.js

```diff
@@ -0,0 +1,168 @@
+import React from 'react';
+import moment from 'moment';
+import { makeStyles } from '@material-ui/core/styles';
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import { KeyboardTimePicker } from '@material-ui/pickers';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogActions from '@material-ui/core/DialogActions';
+import ScheduleIcon from '@material-ui/icons/Schedule';
+import TextField from '@material-ui/core/TextField';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import { connect } from 'react-redux';
+import gameActions from '../store/actions/game';
+
+const useStyles = makeStyles((theme) => ({
+  contentRow: {
+    display: 'flex',
+    justifyContent: 'space-between',
+    '&>*': {
+      marginLeft: theme.spacing(),
+    },
+    '&>:first-child': {
+      marginLeft: 0,
+    },
+  },
+  input: {
+    width: 0,
+    flexGrow: 1,
+  },
+  unlimited: {
+    marginTop: 'auto',
+  },
+  turnInput: {
+    '&::-webkit-outer-spin-button,&::-webkit-inner-spin-button': {
+      '-webkit-appearance': 'none',
+      margin: 0,
+    },
+    '-moz-appearance': 'textfield',
+  },
+  duration: {
+    marginLeft: theme.spacing(2),
+    marginRight: 'auto',
+  },
+}));
+
+const DEFAULT_TURN_LIMIT = 90;
+
+const getNextHour = () => {
+  const now = moment();
+  return now.add(1, 'hour').startOf('hour');
+};
+
+const getDurationInMinutes = (end) => {
+  if (!end) return null;
+  const now = moment();
+  if (end < now) end = end.clone().add(1, 'day');
+  return Math.round(moment.duration(end.diff(now)).asMinutes());
+};
+
+const StartGameDialog = ({ starting, startGame, setStarting }) => {
+  const classes = useStyles();
+  const [turnLimit, setTurnLimit] = React.useState(DEFAULT_TURN_LIMIT);
+  const [timeLimit, setTimeLimit] = React.useState(null);
+  const [duration, setDuration] = React.useState(null);
+
+  const updateTimeLimit = (limit) => {
+    if (!limit) {
+      setTimeLimit(null);
+      setDuration(null);
+      return;
+    }
+
+    const nextLimit = limit.clone();
+    const diffDays = moment.duration(nextLimit.diff(moment())).asDays();
+    nextLimit.subtract(Math.floor(diffDays), 'days');
+    setTimeLimit(nextLimit);
+    setDuration(getDurationInMinutes(nextLimit));
+  };
+  const setNextHour = () => updateTimeLimit(getNextHour());
+  const setUnlimited = () => updateTimeLimit(null);
+  const addMinutes = (minutes) => () => {
+    updateTimeLimit(timeLimit.clone().add(minutes, 'm'));
+  };
+  const closeDialog = () => setStarting(false);
+
+  React.useEffect(() => {
+    if (!starting) return;
+    setTurnLimit(DEFAULT_TURN_LIMIT);
+    setNextHour();
+  }, [starting]);
+
+  const turnLimitInt = parseInt(turnLimit);
+  const isValid = turnLimitInt >= 10 && turnLimitInt <= 3600 &&
+    ((duration >= 1 && duration <= 1440) || (duration === null));
```

Comment:
Might be good to include the unit in some of these variable names

Reply:
Done.